airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/SSHTunnel.java

```diff
@@ -0,0 +1,272 @@
+      this.method = "NO_TUNNEL";
+    } else {
+      this.method = method;
+    }
+    this.host = host;
+    this.tunnelSshPort = tunnelSshPort;
+    this.user = user;
+    this.sshkey = sshkey;
+    this.password = password;
+    this.remoteDatabaseHost = remoteDatabaseHost;
+    this.remoteDatabasePort = remoteDatabasePort;
+    this.tunnelDatabasePort = tunnelDatabasePort;
+  }
+
+  public static SSHTunnel getInstance(JsonNode config) {
+    JsonNode ourConfig = config.get("tunnel_method");
+    SSHTunnel sshconfig = new SSHTunnel(
+        getConfigValueOrNull(ourConfig, "tunnel_method"),
+        getConfigValueOrNull(ourConfig, "tunnel_host"),
+        getConfigValueOrNull(ourConfig, "tunnel_ssh_port"),
+        getConfigValueOrNull(ourConfig, "tunnel_username"),
+        getConfigValueOrNull(ourConfig, "tunnel_usersshkey"),
+        getConfigValueOrNull(ourConfig, "tunnel_userpass"),
+        getConfigValueOrNull(ourConfig, "tunnel_db_remote_host"),
+        getConfigValueOrNull(ourConfig, "tunnel_db_remote_port"),
+        getConfigValueOrNull(ourConfig, "tunnel_localport"));
+    return sshconfig;
+  }
+
+  static String getConfigValueOrNull(JsonNode config, String key) {
+    return config != null && config.has(key) ? config.get(key).asText() : null;
+  }
+
+  /**
+   * Starts an ssh session; wrap this in a try-finally and use closeTunnel() to close it.
+   *
+   * @throws IOException
+   */
+  public void openTunnelIfRequested() throws IOException {
+    if (shouldTunnel()) {
+      if (tunnelSession != null || sshclient != null) {
+        throw new RuntimeException("SSH Tunnel was requested to be opened while it was already open.  This is a coding error.");
+      }
+      sshclient = createClient();
+      tunnelSession = openTunnel(sshclient);
+    }
+  }
+
+  /**
+   * Closes a tunnel if one was open, and otherwise doesn't do anything (safe to run).
+   */
+  public void closeTunnel() {
+    try {
+      if (shouldTunnel()) {
+        if (tunnelSession != null) {
+          tunnelSession.close();
+        }
+        if (sshclient != null) {
+          sshclient.stop();
+        }
+        tunnelSession = null;
+        sshclient = null;
+      }
+    } catch (Throwable t) {
+      throw new RuntimeException(t);
+    }
+  }
+
+  public boolean shouldTunnel() {
+    return method != null && !"NO_TUNNEL".equals(method);
+  }
+
+  public String getMethod() {
+    return method;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  public String getTunnelSshPort() {
+    return tunnelSshPort;
+  }
+
+  public String getUser() {
+    return user;
+  }
+
+  private String getSSHKey() {
+    return sshkey;
+  }
+
+  private String getPassword() {
+    return password;
+  }
+
+  public String getRemoteDatabaseHost() {
+    return remoteDatabaseHost;
+  }
+
+  public String getRemoteDatabasePort() {
+    return remoteDatabasePort;
+  }
+
+  public String getTunnelDatabasePort() {
+    return tunnelDatabasePort;
+  }
```

Comment:
what's the benefit of exposing all of these getters publicly? should they at least be private? or since these variables are only used internally, can we just reference the fields directly?